README.md

```diff
@@ -218,29 +218,173 @@ Authors should respect the `prefers-reduced-motion` media query by conditionally
+// as the top level document. This portal element's previous content has been
+// returned to it. The user now sees this page as it was before we called
+// activate when we were showing the whole viewport preview with this portal.
+// We now reverse the transition, back from 100vw/100vh to 10vw/10vh.
+portal.addEventListener('restore', async () => {
+  // Reverse the animation from the whole viewport preview.
+  if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+    await portal.animate([{ width: '10vw', height: '10vh' }], { duration: 300 }).finished;
+    portal.hidden = true;
+  }
+});
+
+// If the browser could not restore the portal's contents when the user performed
+// a back-navigation (see above about limitations of restoration), then we need
+// to stop showing the portal element, as it's now empty.
+// Note that a portal element may be discarded for other reasons, such as memory
+// pressure on the system, so it's a good idea to handle discards in general.
+portal.addEventListener('discard', async () => {
+  // If we're showing the preview when it's discarded and the user still sees
+  // the previous content, then we can make discarding look nicer by animating
+  // the portal away. Otherwise, just hide it.
+  if (portal.state === 'epitaph' && !portal.hidden &&
+      !matchMedia('(prefers-reduced-motion: reduce)').matches) {
+    await portal.animate([{ width: '10vw', height: '10vh' }], { duration: 300 }).finished;
+    portal.hidden = true;
+  } else {
+    portal.hidden = true;
+    portal.style.width = '10vw';
+    portal.style.height = '10vh';
+  }
+});
+```
+
+Now let's consider a more advanced example where a user goes back and forth between pages of a site. Suppose we have a page embedding a widget with a portal. When the user taps on it, it's activated so the user can interact with the widget. The widget page adopts the host page and displays it as a background to produce a lightbox style UI. Suppose the user then performs a back navigation followed by a forward navigation.\
+Host page:
+```js
+function createWidget() {
+  let widgetContainer = document.getElementById('widgetContainer');
+  let portal = document.createElement('portal');
+  portal.src = 'widget-page.html';
+  widgetContainer.append(portal);
+  portal.addEventListener('load', () => {
+    widgetContainer.hidden = false;
+  });
+
+  portal.addEventListener('click', async (e) => {
+    e.preventDefault();
+    if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      await portal.animate(/* ... */).finished;
+    }
+
+    await portal.activate();
+
+    // We happen to know that the widget will adopt us, but if we needed to
+    // check, we can test for |portalHost|.
+    if (!window.portalHost)
+      return;
+
+    // The host is adopted now, so make any desired visual changes for being
+    // embedded.
+  });
+
+  portal.addEventListener(‘restore’, () => {
```

Comment:
Here and below there are some evil curly quotes

Reply:
Done.